### ISSUE TITLE
Fix inconsistent `gnorm` semantics and units across minimizer backends

### ISSUE THREAD

**Issue by AdamsMP92** [MaMMoS-project/mumag_matrixfree2#68]
## Problem

The exported `gnorm` does not consistently represent the same quantity.

In `io_utils.py`, it is converted to an energy density:

```python
gnorm_col = data[:, 4] * Kd_ref
c.gnorm = me.Entity("EnergyDensity", gnorm_col, "J/m^3")
```

However, the default Python PCG minimizer stores the norm of the preconditioned direction:

```python
gnorm_inf_smooth = jnp.max(jnp.abs(y))
```

This is not the raw energy-gradient norm, so converting it to `J/m^3` is probably not valid.

The C++ backend returns the raw scaled tangent-gradient norm instead. Therefore, the meaning of `gnorm` currently depends on the selected backend.

## Expected behavior

- Define one canonical meaning for `gnorm`.
- Use the same definition in the Python and C++ backends.
- Store the preconditioned norm separately if needed.
- Only declare `gnorm` as `EnergyDensity` in `J/m^3` if this conversion is valid and documented.
- Add a regression test comparing both backends.

## Relevant files

- `src/tommos/io_utils.py`
- `src/tommos/hysteresis_loop.py`
- `src/tommos/minimizers.py`
- `src/cpp/cpp_mkl_minimizer.cpp`

How it is stored in the mammos_hysteresis.csv files:

```python
# mammos csv v3
Index,MagneticFluxDensity,MagneticPolarisation,EnergyDensity,EnergyDensity
,,,,
https://w3id.org/emmo#EMMO_0cd58641_824c_4851_907f_f4c3be76630c,https://w3id.org/emmo#EMMO_961d1aba_f75e_4411_aaa4_457f7516ed6b,https://w3id.org/emm>
,T,T,J / m3,J / m3
config,B_ext_T,J_par_T,E,gnorm
0.0,2.326419728,1.115972067,-2319599.9643322746,50.50292743225158
0.0,2.314787629,1.115456263,-2309272.3150221235,104.23056567963954
0.0,2.30315553,1.11493496,-2298949.4622067483,9.948546173269925
0.0,2.291523432,1.114406938,-2288631.4679590743,108.03212456223685
0.0,2.279891333,1.113872252,-2278318.3948965278,11.241550975006394
0.0,2.268259234,1.11333028,-2268010.306725532,11.446101553959153
```
